mods/default/functions.lua

```diff
@@ -576,3 +576,118 @@ function default.can_interact_with_node(player, pos)
 
 	return false
 end
+
+
+--
+-- Stone/Ore API
+--
+
+default.registered_stones = {}
+default.registered_ores = {}
+
+local function check_stone_groups(groups_a, groups_b)
+	for fieldname, value in pairs(groups_a) do
+		if groups_b[fieldname] == value then
+			return true
+		end
+	end
+	return false
+end
+
+local function register_stone_with_ore(ore, stone)
+	-- set name, description, texture and groups
+	-- name format: 'modNameStone:stoneName_with_oreName'
+	-- for example: 'morestones:granite_with_iron'
+	local fullname = stone.name .. "_with_" .. ore.name
+
+	-- description format: 'Stonename Orename Ore' -> 'Granite Iron Ore'
+	ore.description = stone.description .. " " .. ore.description .. " Ore"
```

Comment:
If I don't create a new definition table here, the ore can pass any other values to this stone registration (e.g. currently `drop`). I'd like to keep this to make the ore registration more flexible (not as in the Farming API, there you can't even pass a `visual_scale` to the nodes).